pandas/_libs/tslibs/nattype.pyx

```diff
@@ -22,6 +22,8 @@ from pandas._libs.tslibs.util cimport (
     get_nat, is_integer_object, is_float_object, is_datetime64_object,
     is_timedelta64_object)
 
+from pandas._libs.missing cimport C_NA
```

Comment:
@jbrockmendel is this a problem? (the dependency)?

Reply:
I'd like to avoid it if possible.  The property "tslibs doesnt rely on anything except _config" is really nice for reasoning-about-code purposes.

@dsaxton is it feasible to put pd.NA into a higher-level check than checknull_with_nat?

Reply:
Hmm, it may be but I think it would have to be before https://github.com/pandas-dev/pandas/blob/master/pandas/core/arrays/datetimes.py#L1844, do you think that looks possible? There's also the "trick" of not importing `NA` at all and checking for `(val == val) is (val != val)` in `checknull_with_nat`

Reply:
ok, going to merge this as-is. @jbrockmendel or @dsaxton if you can figure a way to decouple this a bit would be great (though since missing is pretty dependency free i actually don't mind this)

Reply:
@dsaxton im having trouble figuring out which usage of checknull_with_nat is the relevant one here, can you point it out?  The only ones I see when grepping are in `libmissing.checknull` (which already checks for C_NA, which this renders redundant), and in `Period.__new__`, which seems like it shouldn't be relevant.

Reply:
looks like i was confusing checknull_with_nat with is_null_datetimelike, never mind

Reply:
(this suggests we may have too many null-checking-like functions)